hooks/post-checkout

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# get all the updated .gitattributes files on this branch
+get_changed_attr_file() {
+    git diff --name-only "$( git merge-base master HEAD )" | grep .gitattributes
+}
+
+# scan all files in the given path and check whether the lfs file uses pointer
+check_lfs_integrity_at() {
+    files="$( git ls-files $1 )"
+
+    while read -r f; do
+        local err
+        err=false
+
+        if [[ -n "$( git check-attr filter $f | grep "filter: lfs" )" ]]; then
+            git cat-file blob "HEAD:$f" 2>1 | \
+            git lfs pointer --stdin --file=$f 2>1 > /dev/null || \
+            { err=true; false; }
```

Comment:
not sure what you are trying to do here, is there a way we can simplify this or add some comments to explain what we are checking?

Reply:
simplified and added comments